mm2src/coins/lp_coins.rs

```diff
@@ -1487,6 +1462,25 @@ pub trait CoinAssocTypes {
     fn parse_signature(&self, sig: &[u8]) -> Result<Self::Sig, Self::SigParseError>;
 }
 
+/// Defines associated types specific to Non-Fungible Tokens (Token Address, Token Id, etc.)
+pub trait NftAssocTypes {
+    type ContractAddress: Send + Sync + fmt::Display;
+    type ContractAddrParseError: fmt::Debug + Send + fmt::Display;
+    type TokenId: ToBytes + Send + Sync;
+    type TokenIdParseError: fmt::Debug + Send + fmt::Display;
+    type ContractType: ToBytes + Send + Sync;
+    type ContractTypeParseError: fmt::Debug + Send + fmt::Display;
+
```

Comment:
why can't we have just one error type that covers both error types?
e.g
```rust
 type Error: fmt::Debug + Send + fmt::Display;
```

Reply:
> why can't we have just one error type that covers all error types?

I followed the approach from `CoinAssocTypes` trait
https://github.com/KomodoPlatform/komodo-defi-framework/blob/76af66892d06e7051d2b6e92818bbf1a463d4dc8/mm2src/coins/utxo.rs#L1051-L1061

I suppose Artem decided to keep error types separately, as for UTXO he had errors from crates and usually our own enum errors already implement From for such types.

However, I agree that we can leave one error type for nft assoc trait, because usually we have to create our own structures/enum for this feature, so we cant avoid From trait implementation.

added one err here https://github.com/KomodoPlatform/komodo-defi-framework/pull/2084/commits/95411dc54ea85d814820fdde90bbcc5985c91d95 also renamed eth error as its related to eth coin.

 > Also, this trait name is better as NftParser or something
 
I would like to disagree, `CoinAssocTypes` and `NftAssocTypes` traits are created to parse specific types associated with Coin or with NFT instances. we dont get the whole Coin or Nft.

Reply:
> I would like to disagree, CoinAssocTypes and NftAssocTypes traits are created to parse specific types associated with Coin or with NFT instances. we dont get the whole Coin or Nft.

doesn't mean it's a good name imo